### PR TITLE
Add checkServerIdentity to request config for http adapter

### PR DESCRIPTION
**Pull request by Annihil**
In order to achieve SSL certificate pinning such a shown [here on the NodeJS doc](https://nodejs.org/api/https.html), I added `checkServerIdentity` to the request option for the nodejs http adapter 🙂

### CODE REVIEW

**Comment by yasuf** [comment]
can you add documentation and tests for this?

**Comment by Annihil** [comment]
All done 👍

File: README.md [hunk 1]
```diff
@@ -397,6 +397,17 @@ These are the available config options for making requests. Only the `url` is re
...
   },
 
+  // `checkServerIdentity` defines a hook to check the server's SSL certificate properties (NodeJS only)
+  checkServerIdentity: function (host, cert) {
+    // Pin the exact certificate fingerprint
+    if (cert.fingerprint !== 'E7:EA:A9:74:E1:A1:FF:FD:A8:FB:59:45:1A:AE:92:32:6B:94:23:3E') {
+      const msg = 'Certificate verification error: ' +
+        `The certificate of '${cert.subject.CN}' ` +
+        'does not match our pinned fingerprint';
+      return new Error(msg);
```
Comment (yasuf) [inline README.md]: this should be `throw new Error` I believe?

File: package.json [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "minimist": "^1.2.0",
     "mocha": "^5.2.0",
     "sinon": "^4.5.0",
+    "ssl-root-cas": "^1.3.1",
```
Comment (yasuf) [inline package.json]: is there a way to avoid adding this dependency?

File: test/unit/adapters/http.js [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+var https = require('https');
 var net = require('net');
 var url = require('url');
 var zlib = require('zlib');
 var assert = require('assert');
 var fs = require('fs');
-var server, proxy;
+var path = require('path');
+require('ssl-root-cas')
+  .inject()
```
Comment (yasuf) [inline test/unit/adapters/http.js]: this method is deprecated in this dependency's documentation https://git.coolaj86.com/coolaj86/ssl-root-cas.js#inject, can we use something else?

**Comment by yasuf** [comment]
also can you add your change to the CHANGELOG.md under a new section, with a title: "Next release:"

File: README.md (hunk 1 above)
Comment (Annihil) [reply README.md]: The correct way seems to be `return` according to [NodeJS' doc](https://nodejs.org/api/https.html#https_https_request_url_options_callback)

File: package.json (hunk 2 above)
Comment (Annihil) [reply package.json]: I figured out a way, it's gone now

File: test/unit/adapters/http.js (hunk 3 above)
Comment (Annihil) [reply test/unit/adapters/http.js]: Good catch, `ssl-root-cas` module is not used anymore anyway

**Comment by Annihil** [comment]
I believe this is good to go

**Comment by Annihil** [comment]
@yasuf any thought?

**Comment by Annihil** [comment]
@yasuf could you review again plz?

**Comment by Annihil** [comment]
@yasuf would you know anyone able to merge that?

**Comment by Annihil** [comment]
@yasuf would you know anyone able to merge that?

**Comment by buddhamagnet** [comment]
@yasuf @Annihil is this PR progressing? I'd like to be able to use ```checkServerIdentity``` in one of my projects without having to switch to the core Node packages.

**Comment by kaloudis** [comment]
this rules. would love to see this merged

**Comment by atlanteh** [comment]
Can we please get this into master?

**Pull request merged**